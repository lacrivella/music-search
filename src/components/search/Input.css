@import url('https://fonts.googleapis.com/css?family=Notable|Thasadith&display=swap');
.Input {
  display: block;
  text-align: center;
  flex-direction: row;
  position: relative;
  align-content: space-between;
  background-image: linear-gradient(to right, #fa709a 0%, #fee140 100%);

  input {
    font-family: 'Thasadith', sans-serif;
    font-size: 1.175rem;
    text-align: center;
    height: 3rem;
    width: 7rem;
    margin: 2rem;
  }

  button {
    font-family: 'Notable', sans-serif;
    align-items: center;
    border-radius: 20%;
    font-size: 1.175rem;
    color: #2a2b2e;
  }
}