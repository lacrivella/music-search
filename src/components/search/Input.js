import React from 'react';
import PropTypes from 'prop-types';
import styles from './Input.css';

function Input({ artist, inputChange, onSearch }) {
  return (
    <section className={styles.Input}>
      <input type="text" placeholder="search..." name="artist" value={artist} onChange={inputChange} />
      <button onClick={onSearch}>âŒ•</button>
    </section>
  );
}

Input.propTypes = {
  artist: PropTypes.string.isRequired,
  inputChange: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired
};

export default Input;
