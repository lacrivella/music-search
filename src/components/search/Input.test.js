import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('Input component', () => {
  it('renders Input', () => {
    const wrapper = shallow(<Input artist="David Bowie" inputChange={() => {}} onSearch={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});