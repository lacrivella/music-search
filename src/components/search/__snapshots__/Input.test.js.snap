// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders Input 1`] = `
<section
  className="Input"
>
  <input
    name="artist"
    onChange={[Function]}
    placeholder="search..."
    type="text"
    value="David Bowie"
  />
  <button
    onClick={[Function]}
  >
    âŒ•
  </button>
</section>
`;
